lib/reek/source/source_locator.rb

```diff
@@ -11,60 +12,56 @@ class SourceLocator
       #
       # paths - a list of paths as Strings
       def initialize(paths)
-        @pathnames = paths.
-          map { |path| Pathname.new(path.chomp('/')) }.
-          flat_map { |pathname| current_directory?(pathname) ? pathname.entries : pathname }
+        @paths = paths.map(&method(:Pathname)).flat_map do |path|
+          current_directory?(path) ? path.entries : path
+        end
       end
 
       # Traverses all paths we initialized the SourceLocator with, finds
       # all relevant ruby files and returns them as a list.
       #
-      # @return [Array<File>] - Ruby files found
+      # @return [Array<Pathname>] - Ruby paths found
```

Comment:
We can, but `SourceLocator` taking paths, analysing and filtering them, then returning the right ones seems like a class with a single responsibility dealing with a collection of one other type. Why would `Files` be better here (other than ‘this is what it returned before’)?